src/snap/metric.cc

```diff
@@ -263,3 +276,96 @@ void Metric::Metric::set_last_advert_tm(system_clock::time_point tp) {
     tm->set_sec(nanos / std::nano::den);
     tm->set_nsec(nanos % std::nano::den);
 }
+
+
+
+Namespace::Namespace(std::vector<std::string> ns){
+    
+    for (auto &string_iterator : ns){
+        this->namespace_elements.push_back(Namespace_Element(string_iterator));
+    }
+}
+Namespace::Namespace(){}
+
+Namespace::~Namespace(){
+
+}
+const Namespace_Element Namespace::operator[] (int index) const{
+    return namespace_elements[index];
+}
+
+Namespace& Namespace::add_static_element(std::string value){
+    this->namespace_elements.push_back(Namespace_Element(value));
+    return *this;
+}
+
+Namespace& Namespace::add_dynamic_element(std::string name ,std::string description ){
+    this->namespace_elements.push_back(Namespace_Element("*",name,description));
+    return *this;
+}
+
+std::vector<Namespace_Element> Namespace::get_namespace_elements() const{
+    return this->namespace_elements;
+}
+
+
+const bool Namespace::is_dynamic() const{
+    bool ret = false;
```

Comment:
Using `ret` var is unnecessary, just call `return true` inside loop and `return false` after loop block.